src/Storages/MergeTree/Compaction/MergeSelectors/SimpleMergeSelector.cpp

```diff
@@ -221,9 +223,6 @@ void selectWithinPartition(
             if (settings.max_parts_to_merge_at_once && end - begin > settings.max_parts_to_merge_at_once)
                 break;
 
-            if (!parts[end - 1].shall_participate_in_merges)
```

Comment:
Where is it now?

Reply:
in a collector -- the logic is very simple: if a part should not be used in merges, then the collector will not return that part